apps/coscrad-frontend/src/components/higher-order-components/resource-page.tsx

```diff
@@ -1,8 +1,20 @@
-import { ResourceType } from '@coscrad/api-interfaces';
+import {
+    CategorizableType,
+    CategorizableTypeToViewModel,
+    ResourceType,
+} from '@coscrad/api-interfaces';
 import { useIdFromLocation } from '../../utils/custom-hooks/use-id-from-location';
-import { fullViewCategorizablePresenterFactory } from '../resources/factories/full-view-categorizable-presenter-factory';
 import { AggregateDetailContainer } from './aggregate-detail-container';
 
+type DetailPresenter<T extends CategorizableType> = (
+    viewModel: CategorizableTypeToViewModel[T]
+) => JSX.Element;
+
+interface ResourcePageProps<T extends CategorizableType> {
+    resourceType: ResourceType;
+    detailPresenterFactory: (categorizableType: T) => DetailPresenter<T>;
+}
+
 /**
  * We may want to extends this to an `AggregatePage` some day, or at least
  * `CategorizablePage`. We aren't too concerned about that right now as our core
```

Comment:
Consider doing the latter right now.

Reply:
Tagged the comment with [this story](https://www.pivotaltracker.com/story/show/184069659).